utils.go

```diff
@@ -497,7 +495,6 @@ func (fd *fd) Close() error {
 	value := int(fd.raw)
 	fd.raw = -1
 
-	fd.Forget()
```

Comment:
Won't removing this cause us to accumulate finalizers for fds that have already been closed?

Reply:
I don't think it would accumulate: worst case the GC triggers the finalizer after the `FD` was closed and nothing happens because the `FD` is already closed ... but you're right, we could skip the finalizer if we have actively closed it already. On it !